lib/IRGen/GenStruct.cpp

```diff
@@ -365,9 +407,109 @@ namespace {
 
     void addToAggLowering(IRGenModule &IGM, SwiftAggLowering &lowering,
                           Size offset) const override {
+      auto &layout = ClangDecl->getASTContext().getASTRecordLayout(ClangDecl);
+
+      forEachNonEmptyBase([&](clang::CharUnits offset, clang::CharUnits size) {
+        lowering.addOpaqueData(offset, offset + size);
+      });
```

Comment:
No, this is inappropriate; the members of the base should be added non-opaquely.  The AggTypeLowering defines a lot of ABI details and is important to get right.

But I think `addTypedData` should be doing that already?

Reply:
All of the ABI details should be based on the clang types, which will not have any opaque members. 

If we can't import a member/base class, we import it as opaque. Here, Swift structs never know that they have a base class. They dispatch out to C++ when they need to do anything with a base class member. So, it should be fine for their base classes to be opaque. 

I can look into using `addTypedData` here so we have more information in the lowered Swift structs, but I don't think this should be necessary. And I don't think this change will have any effect on codegen. (That's not to say we shouldn't do it, it would be good to have more info, I just want to make sure we're on the same page about what's currently happening.)

Reply:
This works well. Changes applied :)